### PR TITLE
Add parents option to `file.mkdir`.

### LINKED ISSUE #3600: file.mkdir to support "recursive" dir creation / parent dir creation
`file.mkdir` doesn't support "recursive"/parent dir creation similar to the linux command "mkdir -p xxx". It would be useful for it to support this functionallity.

```
newfolder = "/srv/hlschunks/#{datetime.year}/#{datetime.month}/#{datetime.day}/#{datetime.hour}/#{datetime.min}"

//Current Method
ignore(process.run("mkdir -p #{newfolder}"))

// Ideal Method
file.mkdir(newfolder)
```

### PR DESCRIPTION
**Pull request by smimram**
Fixes #3600.

### CODE REVIEW

File: src/core/builtins/builtins_files.ml [hunk 1]
```diff
@@ -120,10 +124,16 @@ let _ =
...
       try
-        Unix.mkdir dir perms;
+        let rec recmkdir dir =
+          if not (Sys.file_exists dir) then (
+            recmkdir (Filename.dirname dir);
+            Unix.mkdir dir perms)
+        in
+        if parents then recmkdir dir else Unix.mkdir dir perms;
         Lang.unit
       with _ -> Lang.unit)
```
Comment (toots) [inline src/core/builtins/builtins_files.ml:138]: Should we raise a runtime error instead?

File: src/core/builtins/builtins_files.ml [hunk 2]
```diff
@@ -112,6 +112,10 @@ let _ =
   Lang.add_builtin ~base:file "mkdir" ~category:`File
     ~descr:"Create a directory."
     [
+      ( "parents",
```
Comment (toots) [inline src/core/builtins/builtins_files.ml:115]: I think `recursive` is a more common term, that's also the name of the command line option.

**Review by toots — APPROVED** [review]
Looking good! Two comments.

File: src/core/builtins/builtins_files.ml (hunk 2 above)
Comment (toots) [reply src/core/builtins/builtins_files.ml:115]: Hmm no my bad I got confused with `rm`.

File: src/core/builtins/builtins_files.ml (hunk 1 above)
Comment (smimram) [reply src/core/builtins/builtins_files.ml:138]: Let's keep the simple behavior for now.

**Comment by scottgrobinson** [comment]
Hey @toots @smimram - Sorry for the tag on this. Any chance we can look at merging this into 2.2.5? It missed 2.2.4 (I think?)
Thank you!

**Pull request merged**